rfc/system/5606-json-idl.md

```diff
@@ -0,0 +1,343 @@
+# JSON IDL
+
+**Authors:**
+
+- [@Han-Ru](https://github.com/future-outlier)
+- [@Ping-Su](https://github.com/pingsutw)
+
+## 1 Executive Summary
+
+Use json byte string in protobuf instead of json string to fix int is not supported in protobuf struct.
+
+
+## 2 Motivation
+
+In Flytekit, when handling dataclasses, Pydantic base models, and dictionaries, we store data using JSON strings within struct protobuf. This approach causes issues with integers, as protobuf does not support int types, leading to their conversion to floats. This results in performance issues since we need to recursively iterate through all attributes/keys in dataclasses and dictionaries to ensure floats types are converted to int.
+
+Note: We have more than 10 issues about dict, dataclass and Pydantic.
+
+This feature can solve them all.
+
+## 3 Proposed Implementation
+### Flytekit Example
+#### Before
+```python
+@task
+def t1() -> dict:
+  ...
+  return {"a": 1} -> protobuf Struct {"a": 1.0}
+
+@task
+def t2(a: <dict>):
+  print(a["integer"]) # wrong, will be float point
+```
+#### After
+```python
+Json = "json"
+
+@task
+def t1() -> Annotated[dict, Json]: // Json Byte Strings
```

Comment:
Will the user have to use this function signature? As a user I would want to be able to just use ` ->  dict` and not worry about this.